src/runtime/storage/store-context.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Type} from '../type.js';
+import {AbstractStore} from './abstract-store.js';
+
+export interface StoreContext {
+  context?: StoreContext;
```

Comment:
We need to get these abstractions right - what is a `context` and what is `context.context` ? (chicken.chicken!)

These are so fundamental to Arcs that we cannot afford to land half-baked interfaces without documentation.

Some high level observations:

There are as you pointed out 2 pieces of the context that recipe resolver looks into:
A) "arc / local context" - what will be looked up for `use` handles and slots for example.
B) "context proper" - what will be looked up for `map` and `copy` handles for example.

We are introducing StoreContext as we are resolving recipes in 2 different environments:
1) At runtime for a given arc (e.g. in a web shell) - here we have both (A) and (B).
2) At compile time for Android - here we have only (B).

Also @shans and @sjmiles may have some comments here wrt the context!

Reply:
this merely reflects the existing organization of an Arc (or a Manifest); in that there's some local stuff, then a bunch more stuff that is referenced but not local.

To avoid a protracted exercise in design during what's supposed to be a refactor for improving modularity, I would suggest:
(1) rename `StoreContext` to `ArcOrManifest` or `ResolutionContext` or `ResolutionInfo`
(2) make simple adapters for Arc and Manifest rather than making them implement the interface directly (I see you already have this for Manifest)
(2) rename `context` to `globalInfo`, add a `localInfo`
(3) have `localInfo` map to arc-local data, have `globalInfo` map to arc.context
(4) have `localInfo` map to an empty implementation for Manifests, have `globalinfo` map to the manifest data directly.

Reply:
Going w/ option 1, as it's the simplest. 

Is this alright w/ everyone? @shans @piotrswigon

Reply:
This wasn't a list of options.. (unless you mean what to rename `StoreContext` to)

Reply:
My misunderstanding. I was reading too fast.